docs/_posts/2019-02-19-saving-state.md

```diff
@@ -0,0 +1,109 @@
+
+```kotlin
+val sessionStorage SessionStorage(applicationContext, engine)
+
+val sessionManager = sessionManager(engine).apply {
+    sessionStorage.restore()?.let { snapshot -> restore(snapshot) }
+}
+```
+
+ℹ️ Since restoring the last state requires a disk read, it is recommended to perform it off the main thread. This requires the app to gracefully handle the situation where the app starts without any sessions at first. [SessionManager](https://mozac.org/api/mozilla.components.browser.session/-session-manager/) will invoke [onSessionsRestored()](https://mozac.org/api/mozilla.components.browser.session/-session-manager/-observer/on-sessions-restored.html) on a registered [SessionManager.Observer](https://mozac.org/api/mozilla.components.browser.session/-session-manager/-observer/) after restoring has completed.
+
+### SessionBundleStorage
+
+Other than [SessionStorage](https://mozac.org/api/mozilla.components.browser.session.storage/-session-storage/) the [SessionBundleStorage](https://mozac.org/api/mozilla.components.feature.session.bundling/-session-bundle-storage/) implementation can save and restore from multiple states. State is saved as a Bundle in a database.
+
+The storage is setup with a bundle lifetime. [SessionBundleStorage](https://mozac.org/api/mozilla.components.feature.session.bundling/-session-bundle-storage/) will only restore the last bundle if its lifetime has not expired. If there's no active bundle then a new empty bundle will be created to save the state.
+
+```kotlin
+val engine: Engine = ...
+
+val sessionStorage = SessionBundleStorage(
+    applicationContext,
+    bundleLifetime = Pair(1, TimeUnit.HOURS)
+
+val sessionManager = sessionManager(engine).apply {
+    // We launch a coroutine on the main thread. Once a snapshot has been restored
+    // we want to continue with it on the main thread.
+    GlobalScope.launch(Dispatchers.Main) {
+        // We restore on the IO dispatcher to not block the main thread:
+        val snapshot = async(Dispatchers.IO) {
+            val bundle = sessionStorage.restore()
+            // If we got a bundle then restore the snapshot from it
+            bundle.restoreSnapshot(engine)
+        }
+
+        // If we got a snapshot then restore it now:
+        snapshot.await()?.let { sessionManager.restore(it) }
+    }
+}
+```
+
+The storage comes with an [API](https://mozac.org/api/mozilla.components.feature.session.bundling/-session-bundle-storage/#functions) that allows apps to build UI to list, restore, save and remove bundles.
+
+### AutoSave
```

Comment:
nit: I'd find it easier to read if we start with saving, the explain restoring...but not worth changing if this is ready to go. :)